memq-actor/src/main/java/io/appform/memq/actor/Actor.java

```diff
@@ -150,232 +208,308 @@ private ActorObserver setupObserver(List<ActorObserver> observers) {
-                        }
-                        continue;
-                    }
-                    inFlight.addAll(newMessageIds);
-                    val messagesToBeDelivered = newMessageIds.stream()
-                                    .map(messages::get)
-                                            .toList();
-                    messagesToBeDelivered.forEach(internalMessage -> actor.executorService.submit(() -> {
-                        val id = internalMessage.getId();
-                        try {
-                            val observerMessageMeta = new ObserverMessageMeta(id, internalMessage.getPublishedAt(),
-                                    internalMessage.getValidTill());
-                            actor.rootObserver.execute(ActorObserverContext.builder()
-                                            .messageMeta(observerMessageMeta)
-                                            .message(internalMessage.getMessage())
-                                            .operation(ActorOperation.CONSUME)
-                                            .actorName(actor.name)
-                                            .build(),
-                                    () -> process(internalMessage));
-                        }
-                        catch (Throwable throwable) {
-                            log.error("Error processing internalMessage", throwable);
-                        }
-                        finally {
-                            releaseMessage(id);
-                        }
-                    }));
-                }
-            }
-            catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                log.warn("Monitor thread stopped for {}", name);
+                inFlight.remove(id);
+                messages.remove(id);
             }
             finally {
-                lock.unlock();
+                releaseLock();
             }
         }
+    }
 
-        private boolean process(final InternalMessage<M> internalMessage) {
-            val id = internalMessage.getId();
-            val message = internalMessage.getMessage();
-            var status = false;
-            var messageMeta = new MessageMeta(internalMessage.getPublishedAt(),
-                    internalMessage.getValidTill(),
-                    internalMessage.getHeaders());
-            try {
-                val valid = actor.rootObserver.execute(ActorObserverContext.builder()
-                                .message(message)
-                                .operation(ActorOperation.VALIDATE)
-                                .actorName(actor.name)
-                                .build(),
-                        () -> actor.validationHandler.apply(message, messageMeta));
-                if (!valid) {
-                    log.debug("Message validation failed for message: {}", message);
-                    return false;
+    interface Dispatcher<M extends Message> {
```

Comment:
Shall we move this and other implementation classes to separate java files in order to make it more readable ?